src/components/AddEditMovieDialog/AddEditMovieDialog.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useReducer, useState } from "react";
+import { Checkbox, DatePicker, Select } from "antd";
+import dayjs from "dayjs";
+
+import styles from "./AddEditMovieDialog.module.scss";
+import {
+  Input,
+  InputLabels,
+  InputPlaceholders,
+  Dialog,
+  DialogButtonTexts,
+  DialogTitles,
+  datepickerFormat,
+} from "@shared/components";
+import {
+  IMovieInfo,
+  movieGenres,
+} from "@components/MovieList/MovieCard/MovieCard.types";
+import {
+  FormAction,
+  FormState,
+  IAddEditMovieDialogProps,
+} from "./AddEditMovieDialog.types";
+import { formatMinutes } from "@shared/helpers";
+
+const { formContainer, formRow, fieldContainerWithLabel, datePickerAntd } =
+  styles;
+
+const AddEditMovieDialog: React.FC<IAddEditMovieDialogProps> = ({
+  onCancel,
+  onSubmit,
+  isEditModal = false,
+  movieData,
+}) => {
+  const editedMovieData = movieData as IMovieInfo;
+
+  const [runtimeDisplayValue, setRuntimeDisplayValue] = useState<string>(
+    isEditModal ? formatMinutes(+editedMovieData.duration) : "",
+  );
+
+  const initialState: FormState = {
+    title: isEditModal ? editedMovieData.name : "",
+    releaseDate: isEditModal ? editedMovieData.releaseDate : "",
+    imageUrl: isEditModal ? editedMovieData.imageUrl : "",
+    rating: isEditModal ? editedMovieData.rating : "",
+    description: isEditModal ? editedMovieData.description : "",
+    genres: isEditModal ? editedMovieData.genres : [],
+    runtime: isEditModal ? editedMovieData.duration : "",
+  };
+
+  const formReducer = (state: FormState, action: FormAction): FormState => {
+    switch (action.type) {
+      case "SET_TITLE":
+        return { ...state, title: action.payload };
+      case "SET_RELEASE_DATE":
+        return { ...state, releaseDate: action.payload };
+      case "SET_IMAGE_URL":
+        return { ...state, imageUrl: action.payload };
+      case "SET_RATING":
+        return { ...state, rating: action.payload };
+      case "SET_DESCRIPTION":
+        return { ...state, description: action.payload };
+      case "SET_GENRES":
+        return { ...state, genres: action.payload };
+      case "SET_RUNTIME":
+        return { ...state, runtime: action.payload };
+      default:
+        return state;
+    }
```

Comment:
In some article as far as I remember I saw something like this:
if your case name will be the same as properrty it can be look like this:
`
const formReducer = (state: FormState, action: FormAction): FormState => {
        return { ...state, ACTION_TYPE: action.payload };
    }
`
because you repeat all lines: `{ ...state, runtime: action.payload };` only name of the field changed.

Reply:
`enum Types = {
 SET_TITLE: 'title'
SET_IMAGE: 'image'
}`
after that you can use this, if I am not mistakes